### ISSUE TITLE
Error Invalid dimension for content attention

### ISSUE THREAD

**Issue by omprakashsonie** [facebookresearch/wav2letter#662]
I am training for SOTA librispeech with Transformer CTC - it works fine.

For seq2seq, I am getting above error. Pls find below some details. Let me know if you need any more inputs.

Command:
nohup mpirun -n 4 --allow-run-as-root  /root/wav2letter/build/Train train --flagsfile /root/wav2letter/recipes/models/sota/2019/librispeech/train_am_transformer_s2s.cfg --enable_distributed true --minloglevel=0 --logtostderr=1 &

============================
Partial Log:
:
:
:
I0523 12:52:21.388986 31882 Train.cpp:150] Experiment runidx: 1
I0523 12:52:21.391233 31885 Train.cpp:196] Number of classes (network): 9998
I0523 12:52:21.392135 31883 Train.cpp:196] Number of classes (network): 9998
I0523 12:52:21.393010 31884 Train.cpp:196] Number of classes (network): 9998
I0523 12:52:21.393945 31882 Train.cpp:196] Number of classes (network): 9998
I0523 12:52:22.041043 31884 Train.cpp:203] Number of words: 89612
I0523 12:52:22.050370 31882 Train.cpp:203] Number of words: 89612
I0523 12:52:22.058071 31883 Train.cpp:203] Number of words: 89612
I0523 12:52:22.064560 31885 Train.cpp:203] Number of words: 89612
I0523 12:52:22.100217 31882 Train.cpp:217] Loading architecture file from /root/wav2letter/recipes/models/sota/2019/am_arch/am_transformer_ctc.arch
:
:
I0523 12:52:23.022822 31882 Train.cpp:250] [Network Params: 322856814]
I0523 12:52:23.022862 31882 Train.cpp:251] [Criterion] TransformerCriterion
I0523 12:52:23.431706 31882 Train.cpp:259] [Network Optimizer] Adagrad (epsilon=1e-08)
I0523 12:52:23.431751 31882 Train.cpp:260] [Criterion Optimizer] Adagrad (epsilon=1e-08)
I0523 12:52:23.901283 31883 W2lListFilesDataset.cpp:141] 28539 files found.
I0523 12:52:24.019248 31885 W2lListFilesDataset.cpp:141] 28539 files found.
I0523 12:52:24.049134 31884 W2lListFilesDataset.cpp:141] 28539 files found.
I0523 12:52:24.050204 31882 W2lListFilesDataset.cpp:141] 28539 files found.
I0523 12:52:25.145751 31883 W2lListFilesDataset.cpp:141] 104014 files found.
I0523 12:52:25.285117 31885 W2lListFilesDataset.cpp:141] 104014 files found.
I0523 12:52:25.297539 31884 W2lListFilesDataset.cpp:141] 104014 files found.
I0523 12:52:25.325593 31882 W2lListFilesDataset.cpp:141] 104014 files found.
[cc5a494d96e2:31877] 3 more processes have sent help message help-mpi-btl-base.txt / btl:no-nics
[cc5a494d96e2:31877] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
I0523 12:52:26.761293 31883 W2lListFilesDataset.cpp:141] 148688 files found.
I0523 12:52:26.764506 31883 Utils.cpp:102] Filtered 0/281241 samples
I0523 12:52:26.808760 31883 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 8789
I0523 12:52:26.943176 31885 W2lListFilesDataset.cpp:141] 148688 files found.
I0523 12:52:26.946378 31885 Utils.cpp:102] Filtered 0/281241 samples
I0523 12:52:26.991169 31885 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 8789
I0523 12:52:27.030616 31884 W2lListFilesDataset.cpp:141] 148688 files found.
I0523 12:52:27.033440 31884 Utils.cpp:102] Filtered 0/281241 samples
I0523 12:52:27.064283 31882 W2lListFilesDataset.cpp:141] 148688 files found.
I0523 12:52:27.067469 31882 Utils.cpp:102] Filtered 0/281241 samples
I0523 12:52:27.077118 31884 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 8789
I0523 12:52:27.099558 31883 W2lListFilesDataset.cpp:141] 2703 files found.
I0523 12:52:27.099637 31883 Utils.cpp:102] Filtered 0/2703 samples
I0523 12:52:27.099968 31883 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 85
I0523 12:52:27.112298 31882 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 8789
I0523 12:52:27.284204 31885 W2lListFilesDataset.cpp:141] 2703 files found.
I0523 12:52:27.284288 31885 Utils.cpp:102] Filtered 0/2703 samples
I0523 12:52:27.284621 31885 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 85
I0523 12:52:27.329900 31884 W2lListFilesDataset.cpp:141] 2703 files found.
I0523 12:52:27.329979 31884 Utils.cpp:102] Filtered 0/2703 samples
I0523 12:52:27.330338 31884 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 85
I0523 12:52:27.353899 31883 W2lListFilesDataset.cpp:141] 2864 files found.
I0523 12:52:27.353984 31883 Utils.cpp:102] Filtered 0/2864 samples
I0523 12:52:27.354363 31883 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 90
I0523 12:52:27.364526 31882 W2lListFilesDataset.cpp:141] 2703 files found.
I0523 12:52:27.364603 31882 Utils.cpp:102] Filtered 0/2703 samples
I0523 12:52:27.364935 31882 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 85
I0523 12:52:27.535511 31885 W2lListFilesDataset.cpp:141] 2864 files found.
I0523 12:52:27.535586 31885 Utils.cpp:102] Filtered 0/2864 samples
I0523 12:52:27.535941 31885 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 90
I0523 12:52:27.585291 31884 W2lListFilesDataset.cpp:141] 2864 files found.
I0523 12:52:27.585378 31884 Utils.cpp:102] Filtered 0/2864 samples
I0523 12:52:27.585727 31884 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 90
I0523 12:52:27.619626 31882 W2lListFilesDataset.cpp:141] 2864 files found.
I0523 12:52:27.619699 31882 Utils.cpp:102] Filtered 0/2864 samples
I0523 12:52:27.620050 31882 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 90
I0523 12:52:27.818254 31882 Train.cpp:557] Shuffling trainset
I0523 12:52:28.211050 31882 Train.cpp:564] Epoch 1 started!
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid dimension for content attention
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid dimension for content attention
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid dimension for content attention
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid dimension for content attention

============================
configuration file: /root/wav2letter/recipes/models/sota/2019/librispeech/train_am_transformer_s2s.cfg

--runname=am_transformer_seq2seq_librispeech-23may11pm
--rundir=/home/omprakash.s/sota
--archdir=/root/wav2letter/recipes/models/sota/2019
--arch=am_arch/am_transformer_ctc.arch
--tokensdir=/home/omprakash.s/sota/am
--tokens=librispeech-train-all-unigram-10000.tokens
--lexicon=/home/omprakash.s/sota/am/librispeech-train+dev-unigram-10000-nbest10.lexicon
--train=/home/omprakash.s/sota/lists/train-clean-100.lst,/home/omprakash.s/sota/lists/train-clean-360.lst,/home/omprakash.s/sota/lists/train-other-500.lst
--valid=dev-clean:/home/omprakash.s/sota/lists/dev-clean.lst,dev-other:/home/omprakash.s/sota/lists/dev-other.lst
--criterion=transformer
--mfsc
--usewordpiece=true
--wordseparator=_
--am_decoder_tr_dropout=0.1
--am_decoder_tr_layerdrop=0.1
--am_decoder_tr_layers=6
--maxdecoderoutputlen=120
--labelsmooth=0.05
--dataorder=output_spiral
--inputbinsize=25
--attnWindow=softPretrain
--softwstd=4
--trainWithWindow=true
--pretrainWindow=3
--attention=keyvalue
--encoderdim=256
--memstepsize=5000000
--eostoken=true
--pcttraineval=1
--pctteacherforcing=99
--sampletarget=0.01
--netoptim=adagrad
--critoptim=adagrad
--lr=0.03
--lrcrit=0.03
--linseg=0
--momentum=0.0
--maxgradnorm=0.1
--onorm=target
--sqnorm
--nthread=6
--batchsize=8
--filterbanks=80
--minloglevel=0
--logtostderr
--enable_distributed=true

**Comment by vineelpratap** [comment]
Hi, looks like the arch file from your run is different from the original `.cfg` provided in https://github.com/facebookresearch/wav2letter/blob/master/recipes/models/sota/2019/librispeech/train_am_transformer_s2s.cfg .

**Comment by omprakashsonie** [comment]
Thanks Vineel for your help. Closing the issue.

**Comment by rajeevbaalwan** [comment]
Hi @omprakashsonie 
Did you achieved sota result using transformer+CTC model on librispeech ?
